presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -333,14 +340,46 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkTableAnyPermission(ConnectorSecurityContext context, SchemaTableName tableName)
+    {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return true;
+        }
+        if (!isSchemaVisibleToUser(context, tableName.getSchemaName())) {
+            return false;
+        }
+        for (TableAccessControlRule rule : tableRules) {
+            Optional<Set<TablePrivilege>> tablePrivileges = rule.match(context.getIdentity().getUser(), tableName);
+            if (tablePrivileges.isPresent()) {
+                if (tablePrivileges.get().isEmpty()) {
+                    return false;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
     private boolean isSchemaOwner(ConnectorSecurityContext context, String schemaName)
     {
         for (SchemaAccessControlRule rule : schemaRules) {
-            Optional<Boolean> owner = rule.match(context.getIdentity().getUser(), schemaName);
+            Optional<Boolean> owner = rule.isSchemaOwner(context.getIdentity().getUser(), schemaName);
             if (owner.isPresent()) {
                 return owner.get();
             }
         }
         return false;
     }
+
+    private boolean isSchemaVisibleToUser(ConnectorSecurityContext context, String schemaName)
+    {
+        for (SchemaAccessControlRule rule : schemaRules) {
+            Optional<Boolean> visible = rule.isSchemaVisible(context.getIdentity().getUser(), schemaName);
```

Comment:
`return visible.orElse(true)`? 

Because `we assume schema as visible by default`

Reply:
hmm... not necessary, because, if ` (user, schema) ` does not match, we should further look for next one if a match exists